### PR TITLE
Return Space Host Credentials on Bind

### PR DESCRIPTION
**Pull request by micahlee**
Connected to #104

### Description
This PR makes the following changes to the Service Broker:

* Refactored the bind handling to split the different scenarios into separate classes, rather than use conditionals within a single class. This simplifies the bind logic and makes it easier to read for each case ( v4 vs v5, space identity, etc.).

* Added handling of the new space host identity binding. When the broker is configured with the environment variable `ENABLE_SPACE_IDENTITY` set to `true`, the service broker will return a host id and api key for the space, rather than create a host for the application itself. Additionally, if `CONJUR_FOLLOWER_URL` is set, the broker will use the follower for the bind operation, rather than the Conjur master.

### CODE REVIEW

File: app/models/service_binding.rb [hunk 1]
```diff
@@ -1,147 +1,57 @@
...
-
-    return {
-      account: ConjurClient.account,
-      appliance_url: ConjurClient.application_conjur_url,
-      authn_login: "host/#{host_id}",
-      authn_api_key: api_key,
-      ssl_certificate: ConjurClient.ssl_cert || "",
-      version: ConjurClient.version
+  # Look up table for available service bindings by
+  # Conjur version and enabled features (e.g. Space identity)
```
Comment (jtuttle) [inline app/models/service_binding.rb]: This comment seems to indicate that we might create additional indices in this look-up table to handle other features being enabled / disabled? Do we have any plans to do this in the near future? If not we might want to just say explicitly that it's the space identity feature so it's a bit more concrete.

File: features/support/hooks.rb [hunk 2]
```diff
@@ -6,14 +6,26 @@
   skip_this_scenario unless ENV['CONJUR_VERSION'] == "4"
 end
 
+Before("@enable-space-host") do
```
Comment (jtuttle) [inline features/support/hooks.rb]: What happens if you run the cukes without specifying either of these flags? Would you end up w/ a null-ref exception? If so perhaps we should have `@space_host_enabled` default to `false` somewhere and allow this `Before` hook to override it?

File: README.md [hunk 3]
```diff
@@ -95,6 +95,9 @@ To configure the Service Broker to communicate with your external Conjur instanc
 
 - `CONJUR_SSL_CERTIFICATE`: the x509 certificate that was created when Conjur was initiated; this is required for v4 Conjur, but is optional otherwise. If the certificate is stored in a PEM file, you can load it into a local environment variable by calling `export CONJUR_SSL_CERTIFICATE="$(cat tmp/conjur.pem)"`.
 
+- `ENABLE_SPACE_IDENTITY`: When set to the value `true`, the service broker will provide applications with a Space-level `host` identity, rather than create a new `host` identity for the application in Conjur at bind time. This allows the broker
+to use a Conjur follower for application binding, rather than the Conjur master.
```
[the remark this answers is not in the document]
Comment (micahlee) [reply README.md]: Thanks, I will fix it.

File: app/models/service_binding.rb (hunk 1 above)
Comment (micahlee) [reply app/models/service_binding.rb]: Yeah, that makes sense. I guess I was accommodating for future feature flags that may come, but there's no reason to imply that if that don't currently exist :)

File: features/support/hooks.rb (hunk 2 above)
Comment (micahlee) [reply features/support/hooks.rb]: With cukes, the `~` means that this hook matches if the tag *isn't* applied to a scenario. So this is affectively setting the default to false unless the tag is present.

Comment (jtuttle) [reply features/support/hooks.rb]: Oh I see, I thought the `~` case would only trigger if you explicitly provided it in the command to run cucumber, but if that's not the case then looks good to me!

**Pull request merged**